cmd/buildah/config.go

```diff
@@ -100,6 +104,10 @@ func updateConfig(builder *buildah.Builder, c *cli.Context) {
 	if c.IsSet("user") {
 		builder.SetUser(c.String("user"))
 	}
+	if c.IsSet("shell") {
+		shellstr := strings.Replace(c.String("shell"), "\\-", "-", -1)
+		builder.SetShell(strings.Fields(shellstr))
+	}
```

Comment:
Try using `shellwords.Parse()` instead, like we do for `--entrypoint` and `--cmd`.  I think the `strings.Replace()` call won't be necessary then.